### ISSUE TITLE
Evaluation of `fromZ` causes panic

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/saw-script#936]
It seems that evaluation of any expression containing the cryptol function `fromZ` causes the evaluator to panic:
```
sawscript> print {{ fromZ`{13} 2 == 0 }}
saw: You have encountered a bug in SawCore's implementation.
*** Please create an issue at https://github.com/GaloisInc/saw-core/issues

%< ---------------------------------------------------
  Revision:  056ac76c3ba0cdafe0ea054131da8d48660448c9
  Branch:    prettyprinter (uncommited files present)
  Location:  Verifier.SAW.Simulator.Prims
  Message:   expected Integer intEq x
CallStack (from HasCallStack):
  panic, called at src/Verifier/SAW/Utils.hs:37:9 in saw-core-0.1-J19DkSUjZjgBlGUiDiS4Ri:Verifier.SAW.Utils
  panic, called at src/Verifier/SAW/Simulator/Prims.hs:268:13 in saw-core-0.1-J19DkSUjZjgBlGUiDiS4Ri:Verifier.SAW.Simulator.Prims
  panic, called at src/Verifier/SAW/Simulator/Prims.hs:298:15 in saw-core-0.1-J19DkSUjZjgBlGUiDiS4Ri:Verifier.SAW.Simulator.Prims
%< ---------------------------------------------------
```

**Comment by brianhuffman** [comment]
In `Cryptol.sawcore`, the cryptol function `fromZ` is defined in terms of the saw-core primitive `fromIntMod`.

**Comment by brianhuffman** [comment]
I've identified the problem: In the concrete backend, values of the `IntMod` type were originally represented using `VInt`, the same constructor used to represent type `Integer`. The integers are always represented in reduced form, so the original definition of `fromIntMod` was basically just an identity function.

But since GaloisInc/saw-core#93, the `IntMod` type now uses a new, separate value constructor `VIntMod` that also carries the modulus. So the implementation of `fromIntMod` as an identity function is now incorrect.

**Comment by brianhuffman** [comment]
I have a fix for this in GaloisInc/saw-core#109. I'll close the ticket when everything gets merged into saw-script `master`.

**Comment by brianhuffman** [comment]
The fix was merged into saw-script `master` when the saw-core submodule was bumped in #941.
